### ISSUE TITLE
`add_property` does not preserve existing custom properties

### ISSUE THREAD

**Issue by mkaze** [RasaHQ/whatlies#220]
**tl;dr:** As the title says. We should make a copy of embedding in `Embedding.add_property` implementation to resolve this issue.

---
Let's show this in action:

```python
foo = Embedding('foo', [1,2,3])

foo = foo.add_property('x', lambda e: e.vector[0])
print(foo.x)   # works as expected, i.e. prints 1

# let's add another property
foo = foo.add_property('y', lambda e: e.vector[1])
print(foo.y)   # works as expected, i.e. prints 2
print(foo.x)   # raises AttributeError
```

This is also applicable to `EmbeddingSet`:
```python
embset = EmbeddingSet(foo, bar)
embset = embset.add_property('x', lambda e: e.vector[0])
embset['foo'].x   # works as expected
# let's add another property
embset = embset.add_property('y', lambda e: e.vector[1])
embset['foo'].y   # works as expected
embset['foo'].x   # raises AttributeError
```
Well, in both cases, the issue is due to the `add_property` method implementation in `Embedding` class:

https://github.com/RasaHQ/whatlies/blob/681dadf4deb4ecfa3db381974afb21df1fff8ad0/whatlies/embedding.py#L38-L45

As we can see, the existing custom properties are not transferred and preserved in the new `Embedding` instance created. As one solution, making a deep copy instead would resolve the issue.

(relevant: #155, #214)

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @akelad will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by mkaze** [comment]
This was addressed and fixed in #214.
